packages/zowe-explorer/src/trees/job/JobTableView.ts

```diff
@@ -0,0 +1,261 @@
+                        { field: "type", headerName: "Type", filter: true },
+                        { field: "job-correlator", headerName: "Job Correlator", filter: true },
+                        { field: "phase", headerName: "Phase", filter: true },
+                        { field: "phase-name", headerName: "Phase Name", filter: true },
+                        { field: "reason-not-running", headerName: "Error Details", filter: true },
+                        { field: "actions", hide: true },
+                    ]
+                )
+                .addContextOption("all", this.contextOptions.getJcl)
+                .addContextOption("all", this.contextOptions.displayInTree)
+                .addRowAction("all", this.rowActions.cancelJob)
+                .addRowAction("all", this.rowActions.deleteJob)
+                .addRowAction("all", this.rowActions.downloadJob)
+                .build();
+        }
+
+        return this.table;
+    }
+
+    /**
+     * Command handler for the Jobs table view. Called when the action "Show as table" is selected on a Job session node.
+     *
+     * @param context The VS Code extension context (to provide to the table view)
+     * @param node The Job session node that was selected for the action
+     * @param nodeList Passed to `SharedUtils.getSelectedNodeList` to get final list of selected nodes
+     */
+    public static async handleCommand(context: ExtensionContext, node: IZoweJobTreeNode, nodeList: IZoweJobTreeNode[]): Promise<void> {
+        const selectedNodes = SharedUtils.getSelectedNodeList(node, nodeList) as IZoweJobTreeNode[];
+        if (selectedNodes.length !== 1) {
+            return;
+        }
+        if (!SharedContext.isSession(selectedNodes[0])) {
+            return;
+        }
+
+        await this.cacheChildren(selectedNodes[0]);
+        TableViewProvider.getInstance().setTableView(await JobTableView.generateTable(context, selectedNodes[0]));
+    }
+
+    /**
+     * Helper function to obtain all renderable properties from a job node.
+     * @param item The job node to get the properties from
+     * @returns A subset of the `IJob` object, containing renderable properties only
+     */
+    private static jobPropertiesFor(item: IZoweJobTreeNode): Omit<IJob, "step-data"> {
+        return {
+            jobname: item.job.jobname,
+            class: item.job.class,
+            owner: item.job.owner,
+            jobid: item.job.jobid,
+            retcode: item.job.retcode,
+            status: item.job.status,
+            subsystem: item.job.subsystem,
+            type: item.job.type,
+            url: item.job.url,
+            "files-url": item.job["files-url"],
+            "job-correlator": item.job["job-correlator"],
+            phase: item.job.phase,
+            "phase-name": item.job["phase-name"],
+            "reason-not-running": item.job["reason-not-running"],
+        };
```

Comment:
Curious if the following would be equivalent:
```suggestion
        const tempJob = { ...job };
        delete tempJob["step-data"];
        return tempJob;
    }
```

I guess it's probably good to be explicit about which fields we want to display, in case there any other non-serializable fields added by z/OSMF or extenders in the future 😋

Reply:
Yes, it is equivalent - the PoC started with a subset of properties, and once I added more to the object in this function I realized its essentially the same as IJob without `step-data`. Since the API response is unlikely to change, it makes sense to just spread the object and delete that property. This has been resolved in an upcoming commit.